/*
Copyright 2020 The Magma Authors.

This source code is licensed under the BSD-style license found in the
LICENSE file in the root directory of this source tree.

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

syntax = "proto3";

import "lte/protos/subscriberdb.proto";
import "lte/protos/mobilityd.proto";

package magma.lte;
option go_package = "magma/lte/cloud/go/protos";

// --------------------------------------------------------------------------
// [mobilityd] List of assigned IP blocks
// --------------------------------------------------------------------------
message AssignedIPBlocks {
  repeated IPBlock ip_block_list = 1;
}

// --------------------------------------------------------------------------
// [mobilityd] IP descriptor (IP desc) describing an assigned IP
// --------------------------------------------------------------------------
message IPDesc {

  // IPs can be in 1 of 4 states during their lifecycle; see
  // mobilityd.ip_allocator for full description
  enum IPState {
    FREE = 0;
    ALLOCATED = 1;
    RELEASED = 2;
    REAPED = 3;
    RESERVED = 4;
  }

  enum IPType {
    STATIC = 0;
    IP_POOL = 1;
    DHCP = 3;
  }

  IPAddress ip = 1;
  IPBlock ip_block = 2;
  IPState state = 3;
  SubscriberID sid = 4;
  IPType type = 5;
  uint32 vlan_id = 6;
}

// --------------------------------------------------------------------------
// [mobilityd] List of IP descriptors
// --------------------------------------------------------------------------
message IPDescs {
  repeated IPDesc ip_descs = 1;
}
